net/service/sender.go

```diff
@@ -0,0 +1,217 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"github.com/textileio/textile-go/pb"
+	inet "gx/ipfs/QmXfkENeeBvh3zYA51MaSdGUdBjhQ99cP5WQe8zgr6wchG/go-libp2p-net"
+	ggio "gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV/gogo-protobuf/io"
+	"gx/ipfs/QmZoWKhxUmZ2seW4BzX6fJkNR8hh9PsGModr7q171yq2SS/go-libp2p-peer"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+type sender struct {
```

Comment:
`sender`, (no relation to rock climbing), wraps up handling around the i/o streams needed for sending and receiving between each peer